src/components/Profile/RenderFavorites.js

```diff
@@ -0,0 +1,82 @@
+import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import React from "react";
+import { Avatar, ListItem } from "react-native-elements";
+import HandleFavoriteButton from "./HandleFavoriteButton";
+
+const RenderFavorites = ({
+  type,
+  isEditPage = false,
+  items,
+  favorites = [],
+  navigation = null,
+}) => {
+  return (
+    <View>
+      {!isEditPage && (
+        <TouchableOpacity
+          onPress={() => {
+            if (type === "Genre") {
+              navigation.navigate("EditGenre");
+            } else {
+              navigation.navigate("Anime");
+            }
+          }}
+          style={styles.favoriteButton}
+        >
+          <Text style={styles.buttonText}>
+            {type === "Genre"
+              ? "Add more genres to favorite"
+              : "Add more anime to favorite"}
+          </Text>
+        </TouchableOpacity>
+      )}
+
+      {items.map((item, i) => (
+        <ListItem.Swipeable
+          key={i}
+          bottomDivider
+          rightContent={
+            <HandleFavoriteButton
+              type={type}
+              isFavorite={
+                type === "Genre" && isEditPage ? favorites.includes(item) : true
+              }
```

Comment:
type should declared a constant like
`{GENRE: 0, ANIME:1}`
and then you can make another array which refers to the label such as
`["genre", "anime"]`
or something similar